app/src/main/java/com/example/teamproject/MyLocation.java

```diff
@@ -0,0 +1,124 @@
+package com.example.teamproject;
+import android.content.Context;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class MyLocation extends Location{
+    Timer timer1;
+    LocationManager lm;
+    LocationResult locationResult;
+    boolean gps_enabled=false;
+    boolean network_enabled=false;
+
+//    public MyLocation {
+//        //empty constructor
+//    }
+
+
+    public MyLocation(String provider, Timer timer1, LocationManager lm, LocationResult locationResult, boolean gps_enabled, boolean network_enabled, LocationListener locationListenerGps, LocationListener locationListenerNetwork) {
+        super(provider);
+        this.timer1 = timer1;
+        this.lm = lm;
+        this.locationResult = locationResult;
+        this.gps_enabled = gps_enabled;
+        this.network_enabled = network_enabled;
+        this.locationListenerGps = locationListenerGps;
+        this.locationListenerNetwork = locationListenerNetwork;
+    }
+
+
+    public MyLocation(Location l, Timer timer1) {
```

Comment:
Same general comments here: is there a scenario where you pass in these? If not, I would avoid delegating anything that this class should internally handle to its clients.

i.e. If this class can independently manage what timer object to use, you shouldn't have say... your view instantiate a timer object for this class to use. This class can and should do that when it's created!